packages/lexical-yjs/src/Utils.ts

```diff
@@ -541,3 +544,62 @@ export function removeFromParent(node: LexicalNode): void {
     writableNode.__parent = null;
   }
 }
+
+export function $moveSelectionToPreviousParagraph(
+  anchorNodeKey: string,
+  currentEditorState: EditorState,
+) {
+  const currentParagraphKeys = $getRoot().getChildrenKeys();
+  const selectedNode = currentEditorState._nodeMap.get(anchorNodeKey);
+  if (!selectedNode) {
+    $getRoot().selectStart();
+    return;
+  }
+
+  // retrieve the parent paragraph node for the currently selected node
+  const selectedParagraphNode = $findParentParagraphNode(
+    selectedNode,
+    currentEditorState,
+  );
+
+  if (!selectedParagraphNode) {
+    $getRoot().selectStart();
+    return;
+  }
+
+  const prevParagraphNodeKey = selectedParagraphNode.__prev;
+
+  // If collaborator deleted paragraphs including first paragraph
+  // or RootNode accidently set to the selectedParagraphNode
+  if (!prevParagraphNodeKey) {
+    $getRoot().selectStart();
+    return;
+  }
+
+  const prevParagraphNode = $getNodeByKey(prevParagraphNodeKey);
+  if (prevParagraphNodeKey in currentParagraphKeys && prevParagraphNode) {
```

Comment:
As `currentParagraphKeys` is an array, it should probably be `includes()` not `in`
```suggestion
  if (currentParagraphKeys .includes(prevParagraphNodeKey) && prevParagraphNode) {
```

Reply:
thank you!